src/Psalm/Internal/Analyzer/Statements/Block/IfElseAnalyzer.php

```diff
@@ -139,27 +138,24 @@ public static function analyze(
             $if_clauses = [];
         }
 
-        $if_clauses = array_map(
-            /**
-             * @return Clause
-             */
-            function (Clause $c) use ($mixed_var_ids, $cond_object_id): Clause {
-                $keys = array_keys($c->possibilities);
+        $if_clauses_handled = [];
+        foreach ($if_clauses as $clause) {
+            $keys = array_keys($clause->possibilities);
 
-                $mixed_var_ids = array_diff($mixed_var_ids, $keys);
+            $mixed_var_ids = array_diff($mixed_var_ids, $keys);
 
-                foreach ($keys as $key) {
-                    foreach ($mixed_var_ids as $mixed_var_id) {
-                        if (preg_match('/^' . preg_quote($mixed_var_id, '/') . '(\[|-)/', $key)) {
-                            return new Clause([], $cond_object_id, $cond_object_id, true);
-                        }
+            foreach ($keys as $key) {
+                foreach ($mixed_var_ids as $mixed_var_id) {
+                    if (preg_match('/^' . preg_quote($mixed_var_id, '/') . '(\[|-)/', $key)) {
+                        $clause = new Clause([], $cond_object_id, $cond_object_id, true);
                     }
```

Comment:
Shouldn't it have `break 2` here to mimic `return` in the original code?

Reply:
`continue 3` you mean?

Reply:
No, I meant `break 2` as the assignment to `$if_clauses_handles` still needs to happen.

Reply:
I'm a bit scared with this your finding tbh :/ I'll double check the diff today and tomorrow. There are not so much places with multiple returns from a callback and I'll take a closer look into them.